src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/CheckinDetailViewModel.swift

```diff
@@ -11,10 +12,115 @@ final class CheckinDetailViewModel {
 		_ checkin: Checkin
 	) {
 		self.checkin = checkin
+		let components: Set<Calendar.Component> = [.minute, .hour]
+		let eventDuration = Calendar.current.dateComponents(
+			components,
+			from: checkin.traceLocationStartDate ?? Date(),
+			to: checkin.traceLocationEndDate ?? Date()
+		)
+		pickerValues = TwoComponentsIntegerPicker(
+			firstComponentSelectedValue: eventDuration.hour,
+			secondComponentSelectedValue: eventDuration.minute,
+			firstComponentValues: Array(0...23),
+			secondComponentValues: [0, 15, 30, 45]
+		)
 	}
+	
+	// MARK: - Internal
+	
+	let numberOfComponents = 4
+	
+	@OpenCombine.Published var descriptionLabelTitle: String?
+	@OpenCombine.Published var addressLabelTitle: String?
 
```

Comment:
just a comment: I'm using this to update the view as i don't want to make the check-in object public. but im not sure which is better in this case.
Because we already have these strings in the object and there is nothing asynchronous actually happening